src/forms/inputs/file-input/file-input.js

```diff
@@ -114,22 +115,29 @@ class FileInput extends React.Component {
     return (
       <LabeledField { ...this.props }>
         <div className="fileupload fileupload-exists">
-          { 
-            !hidePreview &&
-            renderPreview({ file, value, ...rest })
-          }
-          <div className={ wrapperClass }>
-            <span className="fileupload-exists"> Select File </span>
-              <input 
-                {...{
-                  id: name,
-                  name,
-                  type: 'file',
-                  onChange: this.loadFile,
-                  accept,
-                  'aria-describedby': hasInputError(meta) ? generateInputErrorId(name) : null,
-                }}
-              />
+          {!hidePreview && renderPreview({ file, value, ...rest })}
+          <div>
+            <input
+              {...{
+                id: name,
+                name,
+                type: 'file',
+                onChange: this.loadFile,
+                accept,
+                'aria-labelledby': name + '-label',
+                'aria-describedby': hasInputError(meta)
+                  ? generateInputErrorId(name)
+                  : null,
+              }}
+            />
+            {/* Include after input to allowing for styling with adjacent sibling selector */}
+            <span
+              className={classnames('fileupload-exists', wrapperClass)}
```

Comment:
Taking this `className` at face value, wouldn't we want `fileupload-exists` to be dynamic based upon the state of the component?

Reply:
Hmm, yeah I'm not sure the reasoning behind that naming since it should always be there (following the same logic that was there originally). Maybe it should be renamed to `fileupload-button`?